app/models/throttle.rb

```diff
@@ -48,10 +49,50 @@ class Throttle < ApplicationRecord
     },
   }.freeze
 
+  # Either target or user must be supplied
+  # @param [Symbol] throttle_type
+  # @param [String] target
+  # @param [User] user
+  # @return [Throttle]
+  def self.for(throttle_type:, user: nil, target: nil)
+    throttle = if user
+      find_or_create_by(user: user, throttle_type: throttle_type)
+    elsif target
+      find_or_create_by(target: target, throttle_type: throttle_type)
+    else
+      raise 'Throttle must have a user or a target, but neither were provided'
+    end
+
+    throttle.tap do |t|
+      t.reset_if_expired_and_maxed
+    end
```

Comment:
Style question: Curious why this and not...

```suggestion
    throttle.reset_if_expired_and_maxed
    throttle
```

Reply:
I feel like it was, 6 of one, half a dozen of the other, I will remove `tap` since I used to find that hard to follow as well

Reply:
8db4821535307a30a6963b6325ead9cf7fa7f091